src/hooks/useValidateRepository.ts

```diff
@@ -0,0 +1,188 @@
+import {useCallback} from 'react';
+import {useDebounce} from 'react-use';
+
+import {NamePath} from 'antd/lib/form/interface';
+
+import {dummySecret} from '@utils/sources';
+
+const fieldsNames: string[] = ['uri', 'token', 'username', 'branch', 'path'];
+
+const authSearchString = 'username or token:';
+const uriSearchString = 'uri:';
+const branchSearchString = 'branch:';
+const pathSearchString = 'path:';
+const endSearchString = ', context';
+
+const getErrorMessage = (rawErrorString: string, searchString: string) =>
+  rawErrorString.slice(
+    rawErrorString.search(searchString) + searchString.length,
+    rawErrorString.search(endSearchString)
+  );
+
+const useValidateRepository = (
+  getFieldValue: (name: NamePath) => string,
+  setValidationState: React.Dispatch<any>,
+  validateRepository: any
+) => {
+  const handleResponse = useCallback(
```

Comment:
It doesn't have to be wrapped in `useCallback`, as it's not passed anywhere as dependency (no strict equal check)